src/Controller/ContentManagement/ContentTypeController.php

```diff
@@ -918,7 +930,9 @@ public function editStructureAction($id, Request $request, LoggerInterface $logg
 
             if (array_key_exists('save', $inputContentType) || array_key_exists('saveAndClose', $inputContentType) || array_key_exists('saveAndReorder', $inputContentType)) {
                 $contentType->getFieldType()->updateOrderKeys();
-                $contentType->setDirty($contentType->getEnvironment()->getManaged());
+                $env = $contentType->getEnvironment();
+                $managed = $env ?  $env->getManaged() : false;
```

Comment:
If env is false you should throw a RuntimeException('Unexpected not found environment')